src/ViewModel/ContentHeaderProfile.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace eLife\Patterns\ViewModel;
+
+use Assert\Assertion;
+use eLife\Patterns\ArrayAccessFromProperties;
+use eLife\Patterns\ArrayFromProperties;
+use eLife\Patterns\SimplifyAssets;
+use eLife\Patterns\ViewModel;
+use Traversable;
+
+final class ContentHeaderProfile implements ViewModel
+{
+    use ArrayAccessFromProperties;
+    use ArrayFromProperties;
+    use SimplifyAssets;
+
+    private $details;
+    private $displayName;
+    private $miscLinks;
+    private $logoutLink;
+
+    public function __construct(string $displayName, array $logoutLink, array $miscLinks = [], array $details = [])
```

Comment:
Reuse `Link` for `$logoutLink` (so object instead of array) and `$miscLinks`(array of objects).

Reply:
Should probably have two constructors: one if this is the logged-in user, and one if not.